modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java

```diff
@@ -202,6 +211,29 @@ protected DDMForm getDDMForm(DDMFormInstance ddmFormInstance)
 		return ddmStructure.getDDMForm();
 	}
 
+	private void _createDDMFormFieldOptionsFromDataProvider(
+		ActionRequest actionRequest, DDMForm ddmForm, Locale locale) {
+
+		DDMFormFieldRenderingContext ddmFormFieldRenderingContext =
+			new DDMFormFieldRenderingContext();
+
+		ddmFormFieldRenderingContext.setHttpServletRequest(
+			_portal.getHttpServletRequest(actionRequest));
+		ddmFormFieldRenderingContext.setLocale(locale);
+
+		for (DDMFormField ddmFormField : ddmForm.getDDMFormFields()) {
+			if (Objects.equals(ddmFormField.getType(), "select") &&
+				Objects.equals(
+					ddmFormField.getDataSourceType(), "data-provider")) {
+
+				ddmFormField.setProperty(
+					"options",
+					_ddmFormFieldOptionsFactory.create(
+						ddmFormField, ddmFormFieldRenderingContext));
+			}
+		}
```

Comment:
@marcelabc do we also need to loop through `ddmFormField.getNestedDDMFormFields()` here??

Reply:
Thanks!